src/client/components/CollaboratorAccessModal/CollaboratorAccessModal.tsx

```diff
@@ -0,0 +1,269 @@
+  open: boolean
+}
+
+const CollaboratorAccessModal: React.FC<Props> = (props) => {
+  const { headerLabel, onClose, open, user } = props
+
+  const dispatch = useAppDispatch()
+  const i18n = useTranslation()
+  const assessmentSections = useAssessmentSections()
+
+  const options = assessmentSections
+    .reduce((prev, curr) => [...prev, ...curr.subSections], [])
+    .filter((subSection: SubSection) => subSection.props.anchor)
+    .map(
+      (subSection: SubSection): Option => ({
+        value: subSection.uuid,
+        label: subSection.props.anchor,
+      })
+    )
+
+  const permissionOptions: Record<CollaboratorEditPropertyType, Array<Option>> = {
+    tableData: options,
+    descriptions: options,
+  }
+
+  const properties = (user.roles[0].props as CollaboratorProps) || undefined
+  const sections = Objects.isEmpty(properties) ? 'none' : properties.sections
+
+  const [selectedSections, setSelectedSections] = useState(sections)
+
+  useOnUpdate(() => {
+    dispatch(
+      UserManagementActions.updateSectionAuth({
+        id: user.roles[0].id,
+        sections: selectedSections,
+      })
+    )
+  }, [selectedSections])
+
+  const toggleOptions = (permission: CollaboratorEditPropertyType): void => {
+    const enabled =
+      typeof selectedSections !== 'string'
+        ? Object.entries(selectedSections).filter(([_, section]) => section[permission] === true).length <
+          options.length
+        : true
+    const newSelectedSections = typeof selectedSections !== 'string' ? Objects.cloneDeep(selectedSections) : {}
+    options.forEach((option) => {
+      newSelectedSections[option.value] = { ...newSelectedSections[option.value], [permission]: enabled }
+    })
+    setSelectedSections(newSelectedSections)
+  }
+
+  const removeOption = (section: string, permission: string): void => {
+    if (section === 'all') setSelectedSections('none')
+    else if (section === 'none') setSelectedSections('all')
+    else
+      setSelectedSections(
+        typeof selectedSections !== 'string'
+          ? { ...selectedSections, [section]: { ...selectedSections[section], [permission]: false } }
+          : { [section]: { [permission]: false } }
+      )
+  }
+
+  const addOption = (section: string, permission: CollaboratorEditPropertyType): void => {
+    if (section === 'all') setSelectedSections('all')
+    else if (section === 'none') setSelectedSections('none')
+    else
+      setSelectedSections(
+        typeof selectedSections !== 'string'
+          ? { ...selectedSections, [section]: { ...selectedSections[section], [permission]: true } }
+          : { [section]: { [permission]: true } }
+      )
+  }
+
+  const toggleOption = (section: string, permission: CollaboratorEditPropertyType): void => {
+    if (typeof selectedSections === 'string' || !selectedSections[section]?.[permission]) addOption(section, permission)
+    else if (selectedSections[section]?.[permission]) removeOption(section, permission)
+  }
```

Comment:
export this functions from hook useActions . ```import {toggleOptions...} from './hooks/useActions'```

Reply:
moved logic and state inside useActions